src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -371,6 +375,100 @@ void mutex_init() {
 #undef MUTEX_STORAGE
 #undef MUTEX_STORAGE_NAME
 
+static const int MAX_NAMES = 200;
+static const char* _names[MAX_NAMES] = { nullptr };
+static bool _is_unique[MAX_NAMES] = { false };
+static int _num_names = 0;
+
+PerfCounter** MutexLockerImpl::_perf_lock_count     = nullptr;
+PerfCounter** MutexLockerImpl::_perf_lock_wait_time = nullptr;
+PerfCounter** MutexLockerImpl::_perf_lock_hold_time = nullptr;
+
+void MutexLockerImpl::init_counters() {
```

Comment:
It looks very odd to put an externally callable function in the Impl class. Impl classes are supposed to be private internal details.

Reply:
I've converted `MutexLockerImpl::init_counters()` to a protected function and added a `MutexLocker::init_counters()` which calls `MutexLockerImpl::init_counters()`. In threads.cpp, it can just call `MutexLocker::init_counters()`.